src/ui/modal/modal.interface.ts

```diff
@@ -0,0 +1,25 @@
+import { ComponentType, ReactNode } from 'react'
+import { TransitionStatus } from 'react-transition-group'
+
+export interface IModalComponentProps {
+  state: TransitionStatus
+  transitionDuration: number
+  isMounted: boolean
+}
+
+export type TModalRenderFn = (props: IModalComponentProps) => ReactNode
+
+export interface IModalBaseProps {
+  className?: string
+  transitionDuration: number
```

Comment:
transitionDuration should be optional

Reply:
Yes, you are right. But `Modal` is not an independent component. It's a base for other components that have modal behaviour.